api/src/main/java/org/pmiops/workbench/config/CacheAccessModuleConfiguration.java

```diff
@@ -0,0 +1,33 @@
+package org.pmiops.workbench.config;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.pmiops.workbench.db.dao.AccessModuleDao;
+import org.pmiops.workbench.db.model.DbAccessModule;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.web.context.annotation.RequestScope;
+
+@Configuration
+public class CacheAccessModuleConfiguration {
+  private static final String CAHCE_ACCESS_MODULE_BEAN_NAME = "accessModuleCache";
+
+  @Bean
+  @Qualifier(CAHCE_ACCESS_MODULE_BEAN_NAME)
+  Supplier<List<DbAccessModule>> getCacheAccessModule(AccessModuleDao accessModuleDao) {
+    // Cache configuration in memory for ten minutes.
+    return Suppliers.memoizeWithExpiration(accessModuleDao::findAll, 10, TimeUnit.MINUTES);
```

Comment:
A better pattern for "single cache". 
https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/config/CacheSpringConfiguration.java is multiple cache, make sense to use CacheBuilder Pattern. 
However, 
https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/config/CacheMySqlStopWordConfig.jSupplier pattern is cleaner...

Reply:
I don't understand how spring caching works, so it would be helpful for someone else to review this part

Reply:
I checked this part with CH last Friday, and seems mine seems to be a approach. 
![Screen Shot 2021-07-09 at 11 06 56 AM](https://user-images.githubusercontent.com/6351731/125099212-c7213500-e0a5-11eb-8562-b1551a5d43b6.png)